### PR TITLE
Remove workaround for Sin

### PR DESCRIPTION
**Pull request by umangyadav**
Sine function workaround was added in rocm-5.3 as it was broken for some reason. #1508 
In rocm-5.4.2 it seems to be working again. Removing the workaround to avoid dealing with hip_math_constant.h or HIP_PI_O2 definiton.

### CODE REVIEW

File: src/targets/gpu/kernels/include/migraphx/kernels/hip.hpp [hunk 1]
```diff
@@ -29,9 +29,8 @@
 #include <hip/hip_fp16.h>
 #include <hip/math_functions.h>
 #include <hip/hip_math_constants.h>
-#elif defined(MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS)
-#include <hip/hip_common.h>
-#include <hip/hip_math_constants.h>
+#else
+#define HIP_PIO2_F 1.570796327F
```
[the remark this answers is not in the document]
Comment (umangyadav) [reply src/targets/gpu/kernels/include/migraphx/kernels/hip.hpp:33]: hiprtc headers are not including `hip_math_constants.h`

[the remark this answers is not in the document]
Comment (umangyadav) [reply src/targets/gpu/kernels/include/migraphx/kernels/hip.hpp:33]: On the side not, PI_2 is only used for  computing `sin`. if that is fixed then, we we can remove this completely.

[the remark this answers is not in the document]
Comment (pfultz2) [reply src/targets/gpu/kernels/include/migraphx/kernels/hip.hpp:33]: So this should be a `constexpr` variable.

[the remark this answers is not in the document]
Comment (umangyadav) [reply src/targets/gpu/kernels/include/migraphx/kernels/hip.hpp]: Done.

File: src/targets/gpu/kernels/include/migraphx/kernels/hip.hpp [hunk 2]
```diff
@@ -29,9 +29,8 @@
 #include <hip/hip_fp16.h>
 #include <hip/math_functions.h>
 #include <hip/hip_math_constants.h>
-#elif defined(MIGRAPHX_ENABLE_HIPRTC_WORKAROUNDS)
-#include <hip/hip_common.h>
-#include <hip/hip_math_constants.h>
+#else
+constexpr auto HIP_PIO2_F = 1.570796327F;
```
Comment (pfultz2) [inline src/targets/gpu/kernels/include/migraphx/kernels/hip.hpp:33]: This should use a different name so it doesnt conflict with hip's macro, and it should go in the math.hpp header probably.

Comment (umangyadav) [reply src/targets/gpu/kernels/include/migraphx/kernels/hip.hpp]: Removed workaround for `sin` instead

**Comment by pfultz2** [comment]
This might break for anyone still using rocm 5.3(which I am using on my system).

**Review by pfultz2 — APPROVED** [review]

**Review by kahmed10 — APPROVED** [review]

**Review by kahmed10 — CHANGES_REQUESTED** [review]
Can you add a half sin test?

**Comment by umangyadav** [comment]
> Can you add a half sin test?

It is already present `test_sin_half` and was added in #1508

**Review by kahmed10 — APPROVED** [review]

**Review by turneram — APPROVED** [review]

**Pull request merged**